src/Nri/Ui/PremiumCheckbox/V1.elm

```diff
@@ -22,13 +21,15 @@ import Nri.Ui.Data.PremiumLevel as PremiumLevel exposing (PremiumLevel(..))
     If you get this message, you should show an `Nri.Ui.Premium.Model.view`
 
 -}
-type alias PremiumConfig msg =
+type alias PremiumConfig level msg =
     { label : String
     , id : String
     , selected : Checkbox.IsSelected
     , disabled : Bool
-    , teacherPremiumLevel : PremiumLevel
-    , contentPremiumLevel : PremiumLevel
+    , teacherPremiumLevel : level
+    , contentPremiumLevel : level
+    , isFree : level -> Bool
+    , allowedFor : level -> level -> Bool
```

Comment:
💭 This is a case where I think it's very risky to not label the parameters, since flipping them will case runtime bugs.  Maybe something like `allowedFor: { userLevel : level, checkboxContentLevel : level } -> Bool` ?